tests/AmazonEmailServiceRegionsTests.pas

```diff
@@ -9,42 +9,141 @@ interface
 type
   [TestFixture]
   TAmazonEmailServiceRegionsTests = class
+  private
+    function GetUrlWithProtocolHTTPS(const Endpoint: string): string;
+    function GetUrlWithProtocolIncorrect(const Endpoint: string): string;
   published
     procedure GetServiceURL_ToUSEast_ReturneEndpointEast1;
+    procedure GetServiceURL_ToUSEastWithProtocolHTTPS_ReturneEndpointEast1;
+    procedure GetServiceURL_ToUSEastWithProtocolIncorrect_ReturneEndpointEast1;
     procedure GetServiceURL_ToUSWest_ReturneEndpointWest2;
+    procedure GetServiceURL_ToUSWestWithProtocolHTTPS_ReturneEndpointEast1;
+    procedure GetServiceURL_ToUSWestWithProtocolIncorrect_ReturneEndpointEast1;
     procedure GetServiceURL_ToEUIreland_ReturneEndpointWest1;
+    procedure GetServiceURL_ToEUIrelandWithProtocolHTTPS_ReturneEndpointEast1;
+    procedure GetServiceURL_ToEUIrelandWithProtocolIncorrect_ReturneEndpointEast1;
   end;
```

Comment:
I don't think it is necessary to test with each endpoint, the logic is the same for all, this just makes changes to the code hard to make.